### PR TITLE
[linalg] Add support for `cholesky` decomposition

### PR DESCRIPTION
**Pull request by jon-tow**
Adds basic support for `cholesky` decomposition.

Issue: #562

### CODE REVIEW

**Comment by awav** [comment]
Awesome! @dan-zheng , @saeta

File: Tests/TensorFlowTests/OperatorTests/MathTests.swift [hunk 1]
```diff
@@ -484,6 +484,36 @@ final class MathOperatorTests: XCTestCase {
...
+        // print(tape.gradient(l, x))
+        // ```
+        let x = Tensor<Float>([[[6, 4], [4, 6]], [[2, 6], [6, 20]]])
+        let computedGradient = gradient(at: x) { cholesky($0).sum() }
+        let expectedGradient = Tensor<Float>([
+            [[0.1897575, 0.02154995],
+             [0.02154995, 0.2738613]],
+            [[2.4748755, -0.7071073],
+             [-0.7071073, 0.3535535]]])
+        assertEqual(computedGradient, expectedGradient, accuracy: 1e-5)
```
Comment (awav) [inline Tests/TensorFlowTests/OperatorTests/MathTests.swift]: The accuracy level: is that a rtol or atol?

Comment (awav) [reply Tests/TensorFlowTests/OperatorTests/MathTests.swift]: in any case it is too high

File: Sources/TensorFlow/Operators/Math.swift [hunk 2]
```diff
@@ -2686,6 +2686,33 @@ internal extension Tensor where Scalar: TensorFlowFloatingPoint {
...
+/// part of the input will be used for this operation. The upper-triangular part
+/// will not be read.
+///
+/// The output is a tensor of the same shape as the input
+/// containing the Cholesky decompositions for all input submatrices `[..., :, :]`.
+///
+/// - Parameter input: A tensor of shape `[..., M, M]`.
+@inlinable
+@differentiable(vjp: _vjpCholesky)
+public func cholesky<T: TensorFlowFloatingPoint>(_ x: Tensor<T>) -> Tensor<T> {
```
Comment (awav) [inline Sources/TensorFlow/Operators/Math.swift]: Does it work with batching?

Comment (jon-tow) [reply Sources/TensorFlow/Operators/Math.swift]: Yup, it sure does! Check out some of the shapes used in testing.
https://github.com/tensorflow/swift-apis/blob/c09de96f739aa8bfce0c41e19213bd2d7fa9a4ba/Tests/TensorFlowTests/OperatorTests/MathTests.swift#L488

**Pull request merged**